packages/block-library/src/page-list/edit.js

```diff
@@ -311,8 +267,23 @@ export default function PageListEdit( {
 		setAttributes( { isNested } );
 	}, [ isNested ] );
 
+	let openModal = null;
+	const onModalMount = ( dataFromModal ) => {
+		openModal = dataFromModal.openModal;
+	};
+
 	return (
 		<>
+			<BlockControls group="other">
+				{ allowConvertToLinks && (
+					<ToolbarButton
+						title={ __( 'Edit' ) }
+						onClick={ () => openModal() }
```

Comment:
I don't think this is a solid approach - openModal can remain null and be called in various circumstances. Plus any component using this modal would need this closure to "receive" an internal method of the modal.

The modal should be mounted (hence open) when it's needed and this is in this edit component (open and close modal are things the edit component does, the modal only needs to know if it should show up or not, and that only to have an onClose event, instead of rendering conditionally).

Not sure what is the correct what suggested in https://github.com/WordPress/gutenberg/pull/47922#discussion_r1106374044 but passing props up is basically the opposite of how it should be I think.